### PR TITLE
fix(test-runner-core): evaluate files globs together

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. evaluate test runner `files` config as a glob array

### CODE REVIEW

File: .changeset/lucky-starfishes-attend.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+---
+
+Evaluate `files` as a glob list, rather than adding the result of each string to the file list.
+This allow `files` glob exclude patterns e.g.
+
+```js
+export default {
+  files: [
+    '**/*.spec.ts', // include `.spec.ts` files
+    '!**/*.e2e.spec.ts', // exclude `.spec.ts` files
```
Comment (daKmoR) [inline .changeset/lucky-starfishes-attend.md:12]: shouldn't that be?

```suggestion
    '!**/*.e2e.spec.ts', // exclude `.e2e.spec.ts` files
```

**Comment by bennypowers** [comment]
@manolakis I added you as co-authored-by on the commit. I didn't realize you already opened a PR for this, my bad

**Comment by manolakis** [comment]
No worries. I closed my PR because yours is more complete 👍

The change in collectTestFiles.ts was almost identical 😅

**Comment by bennypowers** [comment]
No idea why this fails in windows but it appears to be unrelated

**Pull request merged**